rust/src/builders/witness_builder.rs

```diff
@@ -113,6 +113,20 @@ pub struct RequiredWitnessSet {
     pub(crate) redeemers: HashSet<RedeemerWitnessKey>,
 }
 
+impl From<RequiredWitnessSet> for Option<RequiredSigners> {
+    fn from(wits: RequiredWitnessSet) -> Self {
+        let mut required_signers = RequiredSigners::new();
+
+        wits.vkeys.union(&wits.bootstraps).for_each(|hash| required_signers.add(hash));
```

Comment:
Remaining signers is a special field inside transactions. It's not just the sum of whatever witness needs to be added. Notably, it's the combination of all the `missing_signers` and `known_signers`

Reply:
When `NativeScriptWitnessInfoKind::Count(num)` is used, the public keys are fake, so the key hashes should not be assign to `required_signers`. How about keep `TransactionBuilder.set_required_signers()` and let the users set it explicitly? It is not mandatory in real use anyway.

Reply:
Rolled back for now.